### PR TITLE
Split Notebook renderers into a separate extension

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #1909 , #7470
Corresponding PR is here https://github.com/microsoft/vscode-notebook-renderers/pull/59

We need to pull in the image renderers as they have comms, and comms can only be done with renderers when both are in the same extension (requirement of VS Code API).

### CODE REVIEW

File: build/webpack/webpack.datascience-ui.config.builder.js [hunk 1]
```diff
@@ -40,6 +40,10 @@ function getEntry(bundle) {
             return {
                 errorRenderer: [`./src/datascience-ui/error-renderer/index.ts`]
             };
+        case 'imageRenderer':
+            return {
+                imageRenderer: ['./src/datascience-ui/image-renderer/index.ts']
```
Comment (DonJayamanne) [inline build/webpack/webpack.datascience-ui.config.builder.js:45]: This renderer adds the save/plot icon for images.

File: src/datascience-ui/image-renderer/index.ts [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+            container.appendChild(div);
+        }
+    } catch (ex) {
+        console.error(`Failed to render mime type ${mimeString}`, ex);
+    }
+}
+
+function createSaveAsButton(outputItem: OutputItem, ctx: RendererContext<unknown>) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const btn = document.createElement('button') as any;
```
Comment (DonJayamanne) [inline src/datascience-ui/image-renderer/index.ts:106]: Using pure JS,  no need of React.
I couldn't use react as we use webpack 4 and we cannot bundle react projects into ESM targets.
It was either upgrade to webpack 5 (complicated & mess) or simplers to just ditch react

File: package.json [hunk 3]
```diff
@@ -9,7 +9,8 @@
         "name": "Microsoft Corporation"
     },
     "extensionPack": [
-        "ms-toolsai.jupyter-keymap"
+        "ms-toolsai.jupyter-keymap",
+        "ms-toolsai.notebook-renderers"
```
Comment (DonJayamanne) [inline package.json:13]: The renderer extension will contain the renderers for vega, plotly, etc

File: package.json [hunk 4]
```diff
@@ -1828,30 +1829,15 @@
...
             },
             {
-                "id": "jupyter-notebook-renderer",
-                "entrypoint": "./out/client_renderer/renderers.js",
-                "displayName": "Jupyter Notebook Renderer",
-                "requiresMessaging": "optional",
+                "id": "jupyter-notebook-image-renderer",
+                "entrypoint": "./out/datascience-ui/imageRenderer/imageRenderer.js",
+                "displayName": "Jupyter Image Renderer",
+                "requiresMessaging": "always",
```
Comment (DonJayamanne) [inline package.json:1835]: Jupyter extension ships with rendere for images so we can have save as / open plotviewer icons

File: src/datascience-ui/image-renderer/index.ts (hunk 2 above)
Comment (IanMatthewHuff) [reply src/datascience-ui/image-renderer/index.ts:109]: Does the esm-webpack-plugin work here? As I recall that's actually what you pointed out to me when I was doing work with the renderers needed to support module loading instead of script loading. We're already using it in our extension for that purpose.

Comment (IanMatthewHuff) [reply src/datascience-ui/image-renderer/index.ts:109]: Actually nevermind, I'd prefer not to use react for something this simple anyways :).

**Review by rchiodo — APPROVED** [review]

File: src/datascience-ui/image-renderer/index.ts [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    btn.style.top = '5px';
+    btn.style.left = '45px';
+    btn.className = 'plotIcon hidden';
+    btn.ariaPressed = 'false';
+    btn.title = 'Save As';
+    btn.ariaLabel = 'Save As';
+    btn.onclick = () => {
+        if (ctx.postMessage) {
+            ctx.postMessage({
+                type: 'openPlot',
```
Comment (IanMatthewHuff) [inline src/datascience-ui/image-renderer/index.ts:158]: One small tweak here that I think we should make. This is just a string to string comparison. Would be nice to hardcode it now that it's in the same project so we don't accidentally change one side and not the other. Also for SaveAs.

File: src/datascience-ui/image-renderer/index.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/datascience-ui/image-renderer/index.ts:109]: > oes the esm-webpack-plugin work here

No it doesn't, i think react has somethign special that makes it not work.

> d, I'd prefer not to use react for something this simple anyways :).

I agree, react is the new jquery, everyone seems to use that for anythign and everything to do with HTML...
The code isn't clean, but i think its readable enough and makes it smaller without react.

**Review by IanMatthewHuff — CHANGES_REQUESTED** [review]
Just one small tweak, would like 'openPlot' and 'saveAs' to be in one location now and not two separate strings.

**Review by IanMatthewHuff — APPROVED** [review]
Just changing to approved as it's just a small tweak I was suggesting.

File: src/datascience-ui/image-renderer/index.ts (hunk 5 above)
Comment (DonJayamanne) [reply src/datascience-ui/image-renderer/index.ts]: Done

**Review by DavidKutu — APPROVED** [review]

File: .vscode/settings.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 // Place your settings in this file to overwrite default and user settings.
 {
     "files.exclude": {
-        "out": true, // set this to true to hide the "out" folder with the compiled JS files
+        "out": false, // set this to true to hide the "out" folder with the compiled JS files
```
Comment (joyceerhl) [inline .vscode/settings.json:4]: Was this change intentional?

File: news/1 Enhancements/1909.md [hunk 7]
```diff
@@ -0,0 +1 @@
+Split Notebook renderers into a seprate extension.
```
Comment (joyceerhl) [inline news/1 Enhancements/1909.md:1]:
```suggestion
Split Notebook renderers into a separate extension.
```

File: .vscode/settings.json (hunk 6 above)
Comment (DonJayamanne) [reply .vscode/settings.json:4]: oops
